src/main.lua

```diff
@@ -1655,18 +1655,22 @@ addEventingHandler("distribute", utils.hasMatchingTag("Action", "Tick"), functio
 	local function tickEpoch(timestamp, blockHeight, hashchain)
 		-- update demand factor if necessary
 		local demandFactor = demand.updateDemandFactor(timestamp)
+		-- distribute rewards for the epoch and increments stats for gateways, this closes the epoch if the timestamp is greater than the epochs required distribution timestamp
 		local distributedEpoch = epochs.distributeRewardsForEpoch(timestamp)
 		if distributedEpoch ~= nil and distributedEpoch.epochIndex ~= nil then
 			tickedRewardDistributions[tostring(distributedEpoch.epochIndex)] =
 				distributedEpoch.distributions.totalDistributedRewards
 			totalTickedRewardsDistributed = totalTickedRewardsDistributed
 				+ distributedEpoch.distributions.totalDistributedRewards
 		end
-
+		-- prune any gateway that has hit the failed 30 consecutive epoch threshold after the epoch has been distributed
+		local prunedGateways = gar.pruneGateways(timestamp)
```

Comment:
This returns more than just gateways:
<img width="520" alt="image" src="https://github.com/user-attachments/assets/d0441702-7a31-457a-98da-731d7c1ea573">

Reply:
ah - right, here we would actually only care about `slashedGateways` and `stakeSlashed`. no delegate stake would be returned (as it was already computed and done by the `prune` that is called before this).

conversely, in `prune` we should not need to account for `stakeSlashed` or `slashedGateways` as that should will happen on this handler, not in the `prune`